universal-application-tool-0.0.1/app/views/questiontypes/RepeaterQuestionRenderer.java

```diff
@@ -36,43 +40,68 @@ public RepeaterQuestionRenderer(ApplicantQuestion question) {
 
   @Override
   public Tag render(Messages messages) {
-    ContainerTag repeaterFields = div().withId(REPEATER_FIELDS_ID);
-    // TODO: add each answer as a repeaterField
-    repeaterFields.with(repeaterField(PLACEHOLDER, Optional.empty(), 0));
+    RepeaterQuestion enumeratorQuestion = question.createEnumeratorQuestion();
+    ImmutableList<String> entityNames = enumeratorQuestion.getEntityNames();
+
+    ContainerTag enumeratorFields = div().withId(ENUMERATOR_FIELDS_ID);
+    int index;
+    for (index = 0; index < entityNames.size(); index++) {
+      enumeratorFields.with(existingEnumeratorField(Optional.of(entityNames.get(index)), index));
+    }
 
     return div()
         .withClasses(Styles.MX_AUTO, Styles.W_MAX)
         .with(
             div()
                 .withClasses(ReferenceClasses.APPLICANT_QUESTION_TEXT)
                 .withText(question.getQuestionText()),
+            div().withId(PLACEHOLDER_ID).withClass(Styles.HIDDEN).withText(PLACEHOLDER),
```

Comment:
question: not sure I understand this, why is the placeholder hidden?

Reply:
This is hacky:

1. This is a hidden div that holds the placeholder text.
2. The template is created outside of the scope of this instance, so the method that makes the template is `static` and has no reference to question-specific admin-configurable placeholder text.
3. The main.ts method that copies the template finds the placeholder div and puts it in the template.

Reply:
Fixed!!! This is much more straightforward now.